src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -822,6 +823,13 @@ bool IsCompilerDiagnosticIncluded(Diagnostic diagnostic)
             }
         }
 
+        /// <summary>
+        /// Extension point to allow inheriting classes to include diagnostics which would otherwise be filtered. The default implementation returns false
+        /// </summary>
+        /// <param name="diagnostic">the diagnostic which is subject to filter processes.</param>
+        /// <returns>return true to exclude a diagnostic, false to leave it up to internal logic.</returns>
+        protected internal virtual bool IsDiagnosticHandled(Diagnostic diagnostic) => false;
```

Comment:
💭 What if we extract the current `IsCompilerDiagnosticIncluded` method to be a virtual method here? Then the derived type could be:

```
protected override bool IsCompilerDiagnosticInclude(Diagnostic diagnostic, CompilerDiagnostics compilerDiagnostics)
{
  if (base.IsCompilerDiagnosticIncluded(diagnostic, compilerDiagnostic))
    return true;

  // return result of fixable diagnostic ID check
}
```

Reply:
Yeah, that was my first thought, but i figured i'd pick a way which can't break the functionality by overriding, so i kept the changes as small as possible. 

I also thought of adding `List<IDiagnosticFilter> DiagnosticFilters` to `AnalyzerTest` so it is composable. I'll change it as per your request though, because anyone who works with this library, probably knows well enough why his tests suddenly stop working after changing the implementation.